# HG changeset patch
# User Rik <rik@octave.org>
# Date 1385528340 28800
#      Tue Nov 26 20:59:00 2013 -0800
# Branch stable
# Node ID 7c0f59b688fcbd523543ab6b740aa0dca30f69a4
# Parent  989ff97a7e8c899d371130582f7b91827c23aed3
configure.ac: Check for Xft library before building native graphics.

* configure.ac: Check for Xft library before building native graphics.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1136,30 +1136,44 @@ if test -n "$OPENGL_LIBS"; then
   if test -n "$warn_freetype"; then
     native_graphics=no
     OCTAVE_CONFIGURE_WARNING([warn_freetype])
   fi
 
   ## Check for fontconfig library
 
   warn_fontconfig=""
-  if test -z "$warn_freetype"; then
+  if test $native_graphics = yes; then
     PKG_CHECK_MODULES(FONTCONFIG, [fontconfig],
       [have_fontconfig=yes
        OPENGL_LIBS="$FONTCONFIG_LIBS $OPENGL_LIBS"
        AC_DEFINE(HAVE_FONTCONFIG, 1, [Define to 1 if fontconfig is present.])],
       [have_fontconfig=no
        warn_fontconfig="Fontconfig library not found.  Native graphics will be disabled."])
   fi
 
   if test -n "$warn_fontconfig"; then
     native_graphics=no
     OCTAVE_CONFIGURE_WARNING([warn_fontconfig])
   fi
 
+  ## Check for Xft library (when using X11)
+
+  warn_xft=""
+  if test $native_graphics = yes && test "$have_x" = yes; then
+    PKG_CHECK_MODULES(XFT, [xft],
+      [AC_DEFINE(HAVE_XFT, 1, [Define to 1 if Xft is present.])],
+      [warn_xft="Xft library not found.  Native graphics will be disabled."])
+  fi
+
+  if test -n "$warn_xft"; then
+    native_graphics=no
+    OCTAVE_CONFIGURE_WARNING([warn_xft])
+  fi
+
   ## Check for FLTK (www.fltk.org) library
 
   AC_ARG_WITH([fltk-prefix], [
     AS_HELP_STRING([--with-fltk-prefix=PFX],
       [prefix where FLTK is installed (optional)])],
     [fltk_prefix="$withval"],
     [fltk_prefix=""])
 
